From 879053d83f772d618ff1d955a61720ebafdb1a4b Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 01 May 2019 21:45:52 +0300
Subject: [PATCH] [SW-0] preboot: configure UART right after start

The BTL settings might not be the best for the UART clock frequency,
so do not count on it and configure UART first thing in the morning.

Change-Id: I16c2426fb48bab7d6e51aca2a5ccc0fd66e1bdc2
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/preboot.c b/common/preboot.c
index 21cbeb4..d9cd96d 100644
--- a/common/preboot.c
+++ b/common/preboot.c
@@ -158,11 +158,13 @@
 
 	regmap_writel(CPU_BOOT_STATUS_IN_PREBOOT, get_cpu_boot_status_reg());
 
+	hl_uart_init();
 	preboot_version();
 	preboot_pcie_init();
 	preboot_delay();
 	preboot_sync_btl();
 	ret = hl_pll_init(&pll_partial_init);
+	/* reinit uart after PLLs configured */
 	hl_uart_init();
 
 	/* Now that UART is ready notify about PLL init */
